### PR TITLE
creating a todos app

### PR DESCRIPTION
**Pull request by 1stIndominus**
[DEMO LINK](https://1stIndominus.github.io/react_todo-app-loading-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,222 +1,164 @@
...
+      .catch(() => isError);
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case 'active':
```
Comment (loralevitska) [inline src/App.tsx:90]: use enum

File: src/App.tsx [hunk 2]
```diff
@@ -1,222 +1,164 @@
...
+              )}
+            />
+          )}
 
-          <form>
+          <form onSubmit={(event) => {
+            event.preventDefault();
+            handleNewTodos();
+            setQuery('');
+          }}
```
Comment (loralevitska) [inline src/App.tsx:125]: Move to separate function  (transfer the function to a variable, and here only write the name of the function)

File: src/App.tsx [hunk 3]
```diff
@@ -1,222 +1,164 @@
...
               data-cy="NewTodoField"
               type="text"
+              value={query}
               ref={newTodoField}
               className="todoapp__new-todo"
               placeholder="What needs to be done?"
+              onChange={(event) => {
+                setQuery(event.target.value);
+              }}
             />
```
Comment (loralevitska) [inline src/App.tsx:137]: same here. Improve it everywhere

File: src/components/Footer/Footer.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  const optionSelected = (event: any) => {
+    setFilter(event.currentTarget.innerHTML.toLocaleLowerCase());
+  };
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {todos.length - completedTodos.length }
+        {' '}
+        items left
```
Comment (loralevitska) [inline src/components/Footer/Footer.tsx:37]:
```suggestion
        {`{todos.length - completedTodos.length }  items left`}
```

**Review by polosanya — APPROVED** [review]
Good job!

**Pull request closed without merging**